observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,194 @@
+        // parse config for global, e.g. "*"
+        globalLog = true;
+        globalParams = new MethodFilterParams(true,
+            currentHeaderBytes, currentMessageBytes);
+      } else if (isServiceGlob(methodOrServicePattern)) {
+        // parse config for a service, e.g. "service/*"
+        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
+        if (perServiceFilters.containsKey(service)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
+          continue;
+        }
+        MethodFilterParams params = new MethodFilterParams(true,
+            currentHeaderBytes, currentMessageBytes);
+        perServiceFilters.put(service, params);
+      } else {
+        // parse pattern for a fully qualified method, e.g "service/method"
+        if (perMethodFilters.containsKey(methodOrServicePattern)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
+          continue;
+        }
+        MethodFilterParams params = new MethodFilterParams(true,
+            currentHeaderBytes, currentMessageBytes);
+        perMethodFilters.put(methodOrServicePattern, params);
+      }
+    }
+    this.globalLog = globalLog;
+    this.globalParams = globalParams;
+    this.perServiceFilters = ImmutableMap.copyOf(perServiceFilters);
+    this.perMethodFilters = ImmutableMap.copyOf(perMethodFilters);
+    this.methodOrServiceFilterPresent = true;
+  }
+
+  @VisibleForTesting
+  void setEventFilterSet() {
+    ImmutableList<EventType> eventFilters = config.getEventTypes();
+    if (eventFilters == null) {
+      return;
+    }
+    this.logEventTypeSet = ImmutableSet.copyOf(eventFilters);
+  }
+
+  static boolean isServiceGlob(String input) {
+    return input.endsWith("/*");
+  }
+
+  public static final class MethodFilterParams {
+    boolean log;
+    int headerBytes;
+    int messageBytes;
+
+    /**
+     * Object used to represent results for service/method filtering.
+     *
+     * @param log logs the corresponding method if set to true
+     * @param headerBytes header bytes limit specified by the config for the method/pattern
+     * @param messageBytes message bytes limit specified by the config for the method/pattern
+     */
+    public MethodFilterParams(boolean log, int headerBytes, int messageBytes) {
+      this.log = log;
+      this.headerBytes = headerBytes;
+      this.messageBytes = messageBytes;
+    }
+  }
+
+  /**
+   * Checks if the corresponding service/method passed needs to be logged as per the user
+   * provided configuration.
+   *
+   * @param fullMethodName the fully qualified name of the method
+   * @return MethodFilterParams object
+   *     1. specifies if the corresponding method needs to be logged (log field will be set to true)
+   *     2. values of payload limits retrieved from configuration
+   */
+  public MethodFilterParams isMethodToBeLogged(String fullMethodName) {
```

Comment:
Wouldn't it be easier/better for this function to just take the `MethodDescriptor` object instead of a String?

Reply:
Updated to take `MethodDescriptor` object instead of a `String`